pkg/sqlmigration/049_add_notification_routes.go

```diff
@@ -0,0 +1,290 @@
+	ExpressionKind string `bun:"kind,type:text" json:"kind"`
+
+	Channels []string `bun:"channels,type:jsonb" json:"channels"`
+
+	Name        string   `bun:"name,type:text" json:"name"`
+	Description string   `bun:"description,type:text" json:"description"`
+	Enabled     bool     `bun:"enabled,type:boolean,default:true" json:"enabled"`
+	Tags        []string `bun:"tags,type:jsonb" json:"tags,omitempty"`
+
+	OrgID string `bun:"org_id,type:text,notnull" json:"orgId"`
+}
+
+type rule struct {
+	bun.BaseModel `bun:"table:rule"`
+	types.Identifiable
+	types.TimeAuditable
+	types.UserAuditable
+	Deleted int    `bun:"deleted,notnull,default:0"`
+	Data    string `bun:"data,type:text,notnull"`
+	OrgID   string `bun:"org_id,type:text"`
+}
+
+type addNotificationRoutes struct {
+	sqlstore  sqlstore.SQLStore
+	sqlschema sqlschema.SQLSchema
+	logger    *slog.Logger
+}
+
+func NewAddNotificationRoutesFactory(sqlstore sqlstore.SQLStore, sqlschema sqlschema.SQLSchema) factory.ProviderFactory[SQLMigration, Config] {
+	return factory.NewProviderFactory(factory.MustNewName("add_notification_routes"), func(ctx context.Context, providerSettings factory.ProviderSettings, config Config) (SQLMigration, error) {
+		return newAddNotificationRoutes(ctx, providerSettings, config, sqlstore, sqlschema)
+	})
+}
+
+func newAddNotificationRoutes(_ context.Context, settings factory.ProviderSettings, _ Config, sqlstore sqlstore.SQLStore, sqlschema sqlschema.SQLSchema) (SQLMigration, error) {
+	return &addNotificationRoutes{
+		sqlstore:  sqlstore,
+		sqlschema: sqlschema,
+		logger:    settings.Logger,
+	}, nil
+}
+
+func (migration *addNotificationRoutes) Register(migrations *migrate.Migrations) error {
+	if err := migrations.Register(migration.Up, migration.Down); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (migration *addNotificationRoutes) Up(ctx context.Context, db *bun.DB) error {
+	// Create the notification_routes table
+	_, err := db.NewCreateTable().
+		Model((*expressionRoute)(nil)).
+		IfNotExists().
+		Exec(ctx)
+	if err != nil {
+		return err
+	}
+
+	indexes := []string{
+		"CREATE INDEX IF NOT EXISTS idx_notification_routes_org_id ON notification_routes (org_id)",
+		"CREATE INDEX IF NOT EXISTS idx_notification_routes_enabled ON notification_routes (enabled)",
+		"CREATE INDEX IF NOT EXISTS idx_notification_routes_org_enabled ON notification_routes (org_id, enabled)",
+		"CREATE INDEX IF NOT EXISTS idx_notification_routes_kind ON notification_routes (kind)",
+		"CREATE INDEX IF NOT EXISTS idx_notification_routes_org_kind ON notification_routes (org_id, kind)",
+		"CREATE INDEX IF NOT EXISTS idx_notification_routes_name ON notification_routes (name)",
+		"CREATE INDEX IF NOT EXISTS idx_notification_routes_org_route_name ON notification_routes (org_id, name)",
+	}
+
+	for _, indexSQL := range indexes {
+		_, err := db.ExecContext(ctx, indexSQL)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Do it in transaction, and see how to create index using `CreateIndex` instead of the pain strings list